### DISCUSSION TITLE
Clicking Heading Link / ToC Heading Link causes the site to freeze for a moment

### DISCUSSION THREAD

**Discussion by Skreeauk** [fuma-nama/fumadocs#591] [Q&A] [answered]
Just as the title, somehow it freezes and no interactivity can be done (scrolling, hovering, clicking, etc) for about 2-3s. Would anyone have any idea as to why this is happening? 

I'm seeing this behaviour on Chrome 126.0.6478.115 & Edge 126.0.2592.81
My site: https://dark-gbf.vercel.app/docs/celeste

**Comment by fuma-nama** [comment, marked as answer]
I can confirm this is caused by your custom animation using the View Transition API, I only observed a `::view-transition` element appeared when clicking heading links

**Reply by Skreeauk** [reply]
Just tested and you're right. Thanks so much for helping.

This makes so much more sense now because I was trying on Mozilla & Safari and it was working fine, just checked and View Transition API isn't supported on Mozilla & Safari yet.

It did not come to my mind that View Transition would be the culprit.

Thanks once again. Cheers
